### ISSUE TITLE
Separate framework code and project-specific code

### ISSUE THREAD

**Issue by szilardd** [abpframework/abp#11000]
Is there any official documentation/tips related to how to keep the core Abp.io framework code and project-specific code separate (as much as possible) in order to avoid lots of conflicts/issues when updating the framework to a newer version?

With the 'old' Zero/AspNetBoilerplate framework our approach was to create separate `AbpModule`s in Class Libraries and reference them in the 'main' projects. This works but there were some places where the complexity increased considerably because of this, especially when having to separate the Entity Framework context across multiple Class Libraries.

This approach doesn't work for example with [Data Seeding](https://docs.abp.io/en/abp/5.0/Data-Seeding). If the `IDataSeedContributor` class is in a Class Library, the `DbMigrator` project doesn't pick it up. It can be solved by using a [Shared Project](https://docs.microsoft.com/en-us/xamarin/cross-platform/app-fundamentals/shared-projects?tabs=windows) instead of Class Library. See related Stack Overflow [question](https://stackoverflow.com/questions/70290082/abp-io-data-seeder-from-external-module).

While Shared Projects have advantages over Class Libraries, they also have disadvantages.

The third option would be to add the project-specific code in the 'main' projects in a separate folder and try to keep changes to the framework code as small as possible.

To summarise, the three options described above:
1. Separate folder in main projects - PRO: no need for separate projects; CON: tight coupling between framework and project code
2. AbpModules as Class Libraries - PRO: keeps code separate; CON: increased complexity in setting up modules
3. Shared Projects referenced in main projects - PRO/CON a mix between options 1 & 2

Wondering which of these would be best, or if there is some other approach others have found. Thanks!

**Comment by iyilm4z** [comment]
There are already solutions in abp for seperating your own code as much as possible, especially the first option.

1. [Module Startup Template](https://docs.abp.io/en/abp/latest/Startup-Templates/Module)
2. [PlugIn-Modules](https://docs.abp.io/en/abp/latest/PlugIn-Modules)

These solutions still don't help?

**Comment by iyilm4z** [comment]
@maliming we may close this issue?
